src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJavaMultilineStringsAst.txt

```diff
@@ -0,0 +1,113 @@
+PACKAGE_DEF -> package [2:0]
+|--ANNOTATIONS -> ANNOTATIONS [2:39]
+|--DOT -> . [2:39]
+|   |--DOT -> . [2:28]
+|   |   |--DOT -> . [2:22]
+|   |   |   |--DOT -> . [2:11]
+|   |   |   |   |--IDENT -> com [2:8]
+|   |   |   |   `--IDENT -> puppycrawl [2:12]
+|   |   |   `--IDENT -> tools [2:23]
+|   |   `--IDENT -> checkstyle [2:29]
+|   `--IDENT -> grammar [2:40]
+`--SEMI -> ; [2:47]
+CLASS_DEF -> CLASS_DEF [4:0]
+|--MODIFIERS -> MODIFIERS [4:0]
+|   `--LITERAL_PUBLIC -> public [4:0]
+|--LITERAL_CLASS -> class [4:7]
+|--IDENT -> InputRegressionJavaMultilineStringsAst [4:13]
+`--OBJBLOCK -> OBJBLOCK [4:52]
+    |--LCURLY -> { [4:52]
+    |--INSTANCE_INIT -> INSTANCE_INIT [6:4]
+    |   `--SLIST -> { [6:4]
+    |       |--VARIABLE_DEF -> VARIABLE_DEF [7:8]
+    |       |   |--MODIFIERS -> MODIFIERS [7:8]
+    |       |   |--TYPE -> TYPE [7:8]
+    |       |   |   `--IDENT -> String [7:8]
+    |       |   |--IDENT -> doubleQuotes [7:15]
+    |       |   `--ASSIGN -> = [7:28]
+    |       |       `--EXPR -> EXPR [7:30]
+    |       |           `--MULTILINE_STRING_LITERAL -> """ "" """ [7:30]
```

Comment:
@romani ping
Should ML string be similar to current string or should it be similar to block comment where we separate start and end.

````
    |--BLOCK_COMMENT_BEGIN -> /* [2:0]
    |  |--COMMENT_CONTENT ->  example  [2:2]
    |  `--BLOCK_COMMENT_END -> */ [2:10]
````

Reply:
In my opinion, it should be similar to existing string literal token

Reply:
it is not easy question ....
https://openjdk.java.net/jeps/355 ... it is even named so special - `Text Blocks` I think it is intentional.
but it is literal - `A text block is a multi-line string literal`.  It is tempting to treat it as original string - but it is not the same.

I think we need 3 AST nodes for this as we have for block comments . Reasons:
- we will have details on there end `"""` is located, so there will be bunch of Checks that validate location of it vs start `"""`, the same as we have for `{` and `}` (ATTENTION in single line string it was not the issue at all)
- I think text block will be extended in future with features that other languages have for margin/indentation/wrapping rules https://www.oreilly.com/library/view/scala-cookbook/9781449340292/ch01s03.html
- It might open possibility for some thirdparty Checks to have their own parser on such text blocks (as we have over javadoc)

Reply:
@romani I chose this naming because of
1. There is already STRING_LITERAL and new token is a similar thing
2. There is a class in checkstyle named TextBlock, it can be confusing

About the triple quotes - it can be added a separate token, but what name can be? because in case of comments it is clear what token is a comment and what is a comment end, but for this syntax start and end token are same.

Reply:
it is better to use same or close terminology to JDK syntax , I found only http://cr.openjdk.java.net/~abuckley/jep355/text-blocks-jls.html , if you know where to find official syntax rules it would be awesome. Or grammar should be kind of copy of what specification is declaring.

TEXT_BLOCK_LITERAL

> There is a class in checkstyle named TextBlock, it can be confusing

we can rename our class to something else if required.

> it can be added a separate token, but what name can be?

good point.
as it is literal - https://docs.oracle.com/javase/specs/jls/se13/html/jls-3.html#jls-3.10-110 `A literal is the source code representation of a value of a primitive type`, so it is not a [lexeme](https://en.wikipedia.org/wiki/Lexical_analysis#Lexeme).  
Ideally:
```
    |--TEXT_BLOCK_LITERAL_BEGIN -> """ 
    |  |--TEXT_BLOCK_CONTENT ->  example 
    |  `--TEXT_BLOCK_LITERAL_END -> """
```
 but can we make something like this ?

Reply:
I will check. I read JEP once again and saw things that require additional work

Reply:
@pbludov , please share your thoughts on this, to make sure we do not miss something.

Reply:
@pbludov , ping.

Reply:
I like the BEGIN/CONTENT/END scheme. If this could be added to `STRING_LITERAL` as well, then adding `TEXT_BLOCK` support to `MultipleStringLiterals` would be a trivial task. But that would be too much breaking change.